game_patch/object/entity.cpp

```diff
@@ -201,6 +203,104 @@ CodeInjection entity_process_pre_hide_riot_shield_injection{
     },
 };
 
+// initialize new AiInfo field used in `entity_damage` to store if on death, the last damage dealt was explosive
+CodeInjection ai_init_ai_info_injection{
+    0x00403022,
+    [](auto& regs) {
+        rf::AiInfo* aip = regs.esi;
+        if (aip) {
+            aip->explosive_last_damage = false;
+        }
+    }
+};
+
+bool consider_gibs() {
+    return g_game_config.gibs && !rf::is_multi;
+}
+
+FunHook<void(int)> entity_blood_throw_gibs_hook{
+    0x0042E3C0, [](int handle) {
+        if (!consider_gibs()) {
+            return;
+        }
+
+        rf::Object* objp = rf::obj_from_handle(handle);
+
+        if (!objp) {
+            return; // invalid object
+        }
+
+        int explode_vclip_index = rf::vclip_lookup("bloodsplat");
+        int chunk_explode_vclip_index = rf::vclip_lookup("bloodsplat");
+        float explode_vclip_radius = 1.0f;
+        const char* debris_filename = "df_meatchunks0.V3D";
+
+        static const int debris_max_lifetime = 7000; // ms
+        static const float debris_velocity = 8.5f;
+        static const float damage_scale = 1.0f;
+        static rf::String cust_snd_set = "gib bounce";
+
+        if (objp->type == rf::OT_ENTITY) { // use overrides from associated entity.tbl class if present
+            rf::Entity* ep = static_cast<rf::Entity*>(objp);
+
+            explode_vclip_index = (ep->info->explode_vclip_index > 0)
+				? ep->info->explode_vclip_index : explode_vclip_index;
+
+            explode_vclip_radius = (ep->info->explode_vclip_radius > 0.0f)
+				? ep->info->explode_vclip_radius : explode_vclip_radius;
+
+            debris_filename = (!ep->info->debris_filename.empty())
+                ? ep->info->debris_filename.c_str() : debris_filename;
+        }
+        else if (objp->type != rf::OT_CORPSE) { // do not gib anything except entities and corpses
+            return;
+        }
+
+        rf::game_do_explosion(
+            explode_vclip_index, objp->room, 0, &objp->pos, explode_vclip_radius, damage_scale, 0);
+
+        rf::debris_spawn_from_object(
+            objp, debris_filename, chunk_explode_vclip_index, debris_max_lifetime, debris_velocity, &cust_snd_set);
+    }
+};
+
+CodeInjection entity_damage_explosive_death_injection{
+    0x0041A413,
+    [](auto& regs) {        
+        rf::Entity* ep = regs.esi;
+        int damage_type = regs.ebp;
+        ep->ai.explosive_last_damage = (damage_type == 3); // explosive damage
```

Comment:
I through I asked about it before but can't find it.
You added a custom flag but one already exists: flag 0x80. It disables some code for custom animations and enable `entity_blood_throw_gibs` call. Not sure why you didn't try to use it.
I've checked PS2 demo and as I expected that flag was really used for gibs there. It was added in `entity_damage` function with following code:
```
    if ((((Player_gore_setting == 1) && ((damaged_ep->__base).life < -100.0)) && ((damaged_ep->__base).material == 3))
       && ((damage_type == 3 || (damage_type == 9)))) {
        damaged_ep->entity_flags = damaged_ep->entity_flags | 0x80;
    }
```
So as I suggested they were basing it on entity material... Even if we add more conditions, I believe we should use already existing flag and also run all the checks here (like V. did), not in 3 places...